### ISSUE TITLE
Merge d9d15af7873fe16d7a0dde4def30f40fa9901777 into 14.0.4

### ISSUE THREAD

**Issue by brad0** [llvm/llvm-project#55577]
[PATCH] [PowerPC] Treat llvm.fmuladd intrinsic as using CTR

This fixes bug [55463](https://github.com/llvm/llvm-project/issues/55463), similar to [D78668](https://reviews.llvm.org/D78668). This is a temporary fix since
we will switch to post-isel CTR loop determination in the future.

**Comment by tstellar** [comment]
@DimitryAndric @chenzheng1030 What do you think about backporting this? https://github.com/llvm/llvm-project/commit/d9d15af7873fe16d7a0dde4def30f40fa9901777

**Comment by chenzheng1030** [comment]
Sounds good to me.

**Comment by DimitryAndric** [comment]
Yeah, that should be fine.

**Comment by tstellar** [comment]
@brad0 The test in this patch is failing on the release/14.x branch.

**Comment by nemanjai** [comment]
@chenzheng1030 Can you create a patch that Tom can apply to the `release/14.x` branch with an update to the test case?

**Comment by chenzheng1030** [comment]
@tstellar Hi, below diff patch should fix the LIT failure.

```
diff --git a/llvm/test/CodeGen/PowerPC/pr55463.ll b/llvm/test/CodeGen/PowerPC/pr55463.ll
index d97306304f25..63d4170da2d5 100644
--- a/llvm/test/CodeGen/PowerPC/pr55463.ll
+++ b/llvm/test/CodeGen/PowerPC/pr55463.ll
@@ -33,10 +33,11 @@ define void @baz() #0 {
 ; CHECK-NEXT:    # kill: def $r7 killed $r7 killed $s7
 ; CHECK-NEXT:    bl fma
 ; CHECK-NEXT:    evmergelo 3, 3, 4
-; CHECK-NEXT:    addi 28, 28, 1
-; CHECK-NEXT:    cmplwi 28, 0
+; CHECK-NEXT:    addi 5, 28, 1
+; CHECK-NEXT:    cmplw 5, 28
+; CHECK-NEXT:    mr 28, 5
 ; CHECK-NEXT:    efdctsiz 3, 3
-; CHECK-NEXT:    bne 0, .LBB0_1
+; CHECK-NEXT:    bge 0, .LBB0_1
 ; CHECK-NEXT:  # %bb.2: # %bb8
 ; CHECK-NEXT:    bl wibble
 ; CHECK-NEXT:    evldd 30, 32(1) # 8-byte Folded Reload
@@ -96,10 +97,11 @@ define void @wombat() #0 {
 ; CHECK-NEXT:    # kill: def $r7 killed $r7 killed $s7
 ; CHECK-NEXT:    bl fma
 ; CHECK-NEXT:    evmergelo 3, 3, 4
-; CHECK-NEXT:    addi 28, 28, 1
-; CHECK-NEXT:    cmplwi 28, 0
+; CHECK-NEXT:    addi 5, 28, 1
+; CHECK-NEXT:    cmplw 5, 28
+; CHECK-NEXT:    mr 28, 5
 ; CHECK-NEXT:    efdctsiz 3, 3
-; CHECK-NEXT:    bne 0, .LBB1_1
+; CHECK-NEXT:    bge 0, .LBB1_1
 ; CHECK-NEXT:  # %bb.2: # %bb8
 ; CHECK-NEXT:    bl wibble
 ; CHECK-NEXT:    evldd 30, 32(1) # 8-byte Folded Reload
```

**Comment by tstellar** [comment]
Merged: 09ec80e16f47
